### PR TITLE
ban command supported

### PR DESCRIPTION
**Pull request by jeevansirela**

### CODE REVIEW

File: files/custom.lua [hunk 1]
```diff
@@ -59,6 +64,12 @@ function auth_on_register(reg)
       auth_token = "GUEST"
     end
     if not decoded["user_uid"] then
+      -- get redis streams
+      -- if inconsistencies remove stream keys
+      -- update banned_users
+      for stream_id, banned_users_table in pairs() do
```
Comment (chetandhembre) [inline files/custom.lua]: not doing anything in for loop.

File: files/custom.lua [hunk 2]
```diff
@@ -135,27 +146,77 @@ function mysplit(str, sep)
...
+function auth_on_subscribe(sub)
+  return true
+end
+
 hooks = {
-  auth_on_register = auth_on_register
+  auth_on_register = auth_on_register,
+  auth_on_publish = auth_on_publish,
+  auth_on_subscribe = auth_on_subscribe
 }
```
Comment (chetandhembre) [inline files/custom.lua]: add new line here

File: files/custom.lua [hunk 3]
```diff
@@ -15,7 +15,12 @@
 -- MySQL Configuration, read the documentation below to properly
 -- provision your database.
 require "auth/auth_commons"
-local jwt = require "auth/luajwt/luajwt" --[[
+local jwt = require "auth/luajwt/luajwt"
+redis.ensure_pool({pool_id = "redis_dev",
```
Comment (chetandhembre) [inline files/custom.lua:19]: what is pool here?

File: files/custom.lua [hunk 4]
```diff
@@ -135,27 +146,77 @@ function mysplit(str, sep)
...
+
+-- redis keys will be deleted from chat server when stream terminates
+function auth_on_publish(pub)
+  local stream_id = string.match(pub.topic, "chat/(.+)/message")
+  if stream_id then
+    local mod_payload, count = string.gsub(pub.payload, "\\n*", "")
+    local decoded_payload = json.decode(mod_payload)
+    local user_uid
+
+    if not decoded_payload and string.len(decoded_payload["message"]) > 240 and not decoded_payload["profile"] then
```
Comment (chetandhembre) [inline files/custom.lua]: `and` condition or `or` condition?

File: files/custom.lua [hunk 5]
```diff
@@ -135,27 +146,77 @@ function mysplit(str, sep)
...
+      return false
+    end
+
+    user_uid = decoded_payload["profile"]["uid"]
```
Comment (chetandhembre) [inline files/custom.lua:187]: what happens when `uid` is not present in message?

File: files/custom.lua [hunk 6]
```diff
@@ -135,27 +146,77 @@ function mysplit(str, sep)
...
+
+    -- verify commands
+    if decoded_payload["profile"]["is_streamer"] then
+      -- if profile_uid is streamer check for ban in message
+      if string.match(decoded_payload["message"], "^/ban ") then
+        local tagged_profile = decoded_payload["tagged_profile"]
+        if not tagged_profile then
```
Comment (chetandhembre) [inline files/custom.lua]: can you explain this if condition?

File: files/custom.lua (hunk 4 above)
Comment (jeevansirela) [reply files/custom.lua]: or only.. sry mistake

File: files/custom.lua (hunk 1 above)
Comment (jeevansirela) [reply files/custom.lua]: supposed to remove

File: files/custom.lua (hunk 6 above)
Comment (jeevansirela) [reply files/custom.lua]: the tagged profile is something we need to ban. whom are we blocking? If it's not present nothing will happen topic (chat/stream_id/command) will receive that message

File: files/custom.lua (hunk 5 above)
Comment (jeevansirela) [reply files/custom.lua:187]: It has to have uid. profile is sender details. I will add that condition.

File: files/custom.lua [hunk 7]
```diff
@@ -135,27 +143,77 @@ function mysplit(str, sep)
...
```
Comment (chetandhembre) [inline files/custom.lua]: just suggestion for code.
```
if tagger_profile and tagger_profile["uid"] then
  cmd(string.format("hmset ban:%s %s %s", stream_id, tagged_profile["uid"], "1"), true)
end
return {topic = string.format("chat/%s/command", stream_id), payload = mod_payload, throttle = 1000 }
```

File: files/custom.lua [hunk 8]
```diff
@@ -135,27 +143,77 @@ function mysplit(str, sep)
...
+          return {topic = string.format("chat/%s/command", stream_id), payload = mod_payload, throttle = 1000 }
+        end
+        cmd(string.format("hmset ban:%s %s %s", stream_id, tagged_profile["uid"], "1"), true)
+        return {topic = string.format("chat/%s/command", stream_id), payload = mod_payload, throttle = 1000 }
+      end
+    else
+      -- if not check if profile uid is banned if yes return false
+      if cmd(string.format("hget ban:%s %s", stream_id, user_uid), "1") then
```
Comment (chetandhembre) [inline files/custom.lua:201]: will return `true` when user_uid present in the hash set?

File: files/custom.lua (hunk 3 above)
Comment (jeevansirela) [reply files/custom.lua:19]: it ensures a connection pool for redis

File: files/custom.lua (hunk 7 above)
Comment (jeevansirela) [reply files/custom.lua]: yea cool..thnks

File: files/custom.lua (hunk 8 above)
Comment (jeevansirela) [reply files/custom.lua:201]: It will forward message to chat/+/command. Message won't be published to chat/+/message

**Review by chetandhembre — APPROVED** [review]

**Pull request closed without merging**